lib/logstash/outputs/elastic_app_search.rb

```diff
@@ -64,13 +65,25 @@ def format_batch(events)
         doc["id"] = event.sprintf(@document_id)
       end
       doc.delete("@version")
-      doc
+      resolved_engine = event.sprintf(@engine)
+      #docs_for_engine[resolved_engine] = [] unless docs_for_engine[resolved_engine]
+      unless docs_for_engine[resolved_engine]
+        if @logger.trace?
+          @logger.trace("Creating new engine segment in batch to send", :resolved_engine => resolved_engine)
+        end
+        docs_for_engine[resolved_engine] = []
+      end
+      eng_name_field = event.get("engine_name_field")
+      docs_for_engine[resolved_engine] << doc
     end
+    docs_for_engine
   end
 
-  def index(documents)
-    response = @client.index_documents(@engine, documents)
-    report(documents, response)
+  def index(batch)
+    batch.each do |resolved_engine, documents|
+      response = @client.index_documents(resolved_engine, documents)
```

Comment:
Without the friendly error checking that we do in `register`/`check_connection!` for a statically defined engine, what do `401` or `404` errors look like to the user?